src/components/LocalIdentityModal/LocalIdentityModal.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import {
+  Button,
+  IdentityBadge,
+  TextInput,
+  breakpoint,
+  font,
+  theme,
+} from '@aragon/ui'
+import { ModalContext } from '../ModalManager/ModalManager'
+import EscapeOutside from '../EscapeOutside/EscapeOutside'
+
+const LocalIdentityModal = ({ opened, ...props }) => {
+  const { showModal, hideModal } = React.useContext(ModalContext)
+  React.useEffect(() => {
+    opened ? showModal(Modal, props) : hideModal()
+  }, [opened])
+
+  return null
+}
+
+LocalIdentityModal.propTypes = {
+  opened: PropTypes.bool.isRequired,
+}
+
+const Modal = ({ address, label, onCancel, onSave }) => {
+  const [action, setAction] = React.useState()
+  const labelInput = React.useRef(null)
+  const handleCancel = () => {
+    onCancel()
+  }
+  const [error, setError] = React.useState(null)
+  const handleSave = () => {
+    try {
+      const label = labelInput.current.value.trim()
+      if (label) {
+        onSave({ address, label })
+      }
+    } catch (e) {
+      setError(e)
+    }
+  }
+  const handleKeyUp = e => {
+    if (e.keyCode === 13) {
+      handleSave()
+    }
+  }
+  React.useEffect(() => {
+    setAction(label && label.trim() ? 'Edit' : 'Add')
+    labelInput.current.focus()
+    labelInput.current.select()
+    window.addEventListener('keyup', handleKeyUp)
+    return () => window.removeEventListener('keyup', handleKeyUp)
+  }, [])
```

Comment:
Shouldn’t we re-run this if `label` changes?

Reply:
The `Modal` is short-lived and so it never gets new props, when the user changes the `label` the `Modal` closes, so no need to run the effect over again.